frontend/src/components/pages/explore/ExploreClubList.tsx

```diff
@@ -0,0 +1,196 @@
+import { ContentWarnings } from "../club_display/ContentWarnings";
+import { createStyles } from "@material-ui/core/styles";
+import Dialog from "@material-ui/core/Dialog";
+import DialogActions from "@material-ui/core/DialogActions";
+import DialogContent from "@material-ui/core/DialogContent";
+import DialogContentText from "@material-ui/core/DialogContentText";
+import DialogTitle from "@material-ui/core/DialogTitle";
+import InputIcon from "@material-ui/icons/Input";
+import { Link } from "react-router-dom";
+import { makeStyles } from "@material-ui/core/styles";
+import PageviewIcon from "@material-ui/icons/Pageview";
+import { Theme } from "@material-ui/core/styles";
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    button : {
+      marginLeft: theme.spacing(1),
+      marginTop: theme.spacing(1),
+      height: '35px',
+    },
+    buttonContainer: {
+      marginTop: theme.spacing(1),
+    },
+    club : {
+      marginTop: theme.spacing(3),
+      width: '900px',
+    },
+    clubContent: {
+      display: 'flex',
+      flexBasis: '100%',
+      flexWrap: 'wrap',
+      marginBottom: theme.spacing(2),
+      marginLeft: theme.spacing(2),
+      marginRight: theme.spacing(2),
+    },
+    clubHeader: {
+      display: 'flex',
+      flexBasis: '100%',
+      justifyContent: 'space-between',
+      marginBottom: theme.spacing(2),
+      marginLeft: theme.spacing(2),
+      marginRight: theme.spacing(2),
+      marginTop: theme.spacing(1),
+    },
+    clubTitle: {
+      marginBottom: theme.spacing(0),
+      maxWidth: '500px',
+    },
+    link: {
+      color: theme.palette.text.primary,
+      textDecoration: 'none',
+    },
+    listedClubsContainer: {
+      alignItems: 'center',
+      display: 'flex',
+      flexDirection: 'column',
+      marginBottom: theme.spacing(3),
+    },
+  }),
+);
+
+interface ExploreClubListProps {
+  clubsToDisplay: ClubInterface[],
+  handleJoinClub(clubId: string): void,
+}
+
+/**
+ * Displays up to the number of clubs that the user has requested to
+ * the page.
+ */
+export function ExploreClubList(props: ExploreClubListProps) {
+  const classes = useStyles();
+  const clubsToDisplay = props.clubsToDisplay;
+
+  const [joinClubAlertOpen, setJoinAlertOpen] = React.useState<boolean>(/* closed= */ false);
+  const [nameOfClubAlert, setNameofClubAlert] = React.useState<string|undefined>(undefined);
+
+  const openAlertWindow = (clubId: string) => {
+    setNameofClubAlert(clubId);
+    setJoinAlertOpen(true);
+  }
+
+  const closeAlertWindow = () => {
+    setJoinAlertOpen(false);
+  }
+
+  const handleJoinClub = (clubId: string) => {
+    props.handleJoinClub(clubId);
+    setJoinAlertOpen(false);
+  }
+
+  /* If clubs to display is not yet defined. */
+  if (clubsToDisplay !== undefined)  {
+    return (
+      <div className={classes.listedClubsContainer}>
+        {clubsToDisplay.map((item, index) => (
+          <div className={classes.club} key={item.name}>
```

Comment:
I think this is fine but it might be worth thinking about how more of this could be shared with the Your Clubs page.

Reply:
I completely agree; ideally I would have /Explore and /YourClubs utilize this as a shared component but between the differences in the two components this design decision was definitely made with time as a leading guide.